lib/galaxy/datatypes/binary.py

```diff
@@ -291,6 +300,14 @@ def set_meta( self, dataset, overwrite = True, **kwd ):
         dataset.metadata.bam_index = index_file
         # Remove temp file
         os.unlink( stderr_name )
+        # Now use pysam with BAI index to determine additional metadata
+        bam_file = csamtools.Samfile( filename=dataset.file_name, mode='rb', index_filename=index_file.file_name )
+        dataset.metadata.bam_header = bam_file.header
+        dataset.metadata.read_groups = [ read_group['ID'] for read_group in dataset.metadata.bam_header.get( 'RG', [] ) if 'ID' in read_group ]
+        dataset.metadata.reference_names = list( bam_file.references )
+        dataset.metadata.reference_lengths = list( bam_file.lengths )
+        dataset.metadata.sort_order = dataset.metadata.bam_header.get( 'HD', {} ).get( 'SO', None )
+        dataset.metadata.bam_version = dataset.metadata.bam_header.get( 'HD', {} ).get( 'VN', None )
```

Comment:
Wow, that was fast!

I know next to nothing about bam files - but I know other bioinformatics formats and parsers - and frequently these two things do not play nice together :). Is that the case for this format and library? If yes - should we wrap this stuff in an exception handler and let Galaxy continue to collect the base metadata for problematic BAMs since these fields are marked as optional anyway? If that is not a realistic concern than I am an enthusiastic :+1:.